design/open/gc-two-steps-separation.md

```diff
@@ -0,0 +1,71 @@
+# Separate GC into two separately runnable steps
+
+## Motivation
+
+Improve the resilience of GC (separation of concerns, pure functions, and traceability) and allow examination
+of intermediate results.
+
+## How?
+
+The `lakefs.debug.gc.no_delete` flag lets users run GC and generate intermediate results,
+yet GC doesn't continue over those results, and retrieving them manually is quite tedious.  
+Both problems can be solved by introducing two flags and changing the `lakefs.debug.gc.no_delete`:
+
+1. `lakefs.debug.gc.no_delete` -> `lakefs.gc.only_mark` (Boolean): This flag instructs the GC operation to only mark, i.e. collect, the addresses that are intended to be deleted.
+2. `lakefs.gc.only_sweep` (Boolean): The GC operation will only sweep, i.e. delete, the marked addresses based on the provided mark ID.
+3. `lakefs.gc.mark_id` (String): This flag specifies which ID the GC will use to generate intermediate results and write outputs to.
+
+## Flags permutations
+
+#### Only Mark
+
+In the case of `lakefs.gc.only_mark`, the GC will run, collect the addresses to be deleted, and write them to:
+`STORAGE_NAMESPACE/_lakefs/retention/gc/addresses/mark_id=RANDOM_VALUE/`.
+
+#### Only Mark and Mark ID
```

Comment:
I think that we should fail GC in case this permutation is used, why would we want to override a previous GC plan run (I'm using the new terminology)? 
thoughts?

Reply:
This is the case where we only want to mark the addresses to be deleted with a specific mark ID so we can examine the results according to it.
Why should it fail here?
As explained in the [Note](https://github.com/treeverse/lakeFS/pull/4254/files#diff-5e7e580bed0e5e9fdb5a1a66fab9a832d7229cbf7ed34596b0a5f22c86ff6eb4R53) below, the users who orchestrate the GC run will need to ensure that they don't override the previous mark/plan or do override it if they wish...

Reply:
It is a very useful flag combination.  I can use it for automation or testing or whatever.

Reply:
I'm ok with it, thanks